server/fleet/app.go

```diff
@@ -132,6 +132,54 @@ type AppConfig struct {
 	didUnmarshalLegacySettings bool
 }
 
+func (c *AppConfig) Clone() (interface{}, error) {
```

Comment:
Is there a bug in https://github.com/fleetdm/fleet/blob/cb0392352b6c3d7c6444726563190f7fdb1827d8/server/datastore/cached_mysql/cached_mysql.go#L33-L69?

In `clone` should we instead use `copier.CopyWithOption(clone.Interface(), v, copier.Option{DeepCopy: true})`?

Reply:
There's no bug _per se_, but this is a shallow clone, i.e. it is the equivalent for a struct pointer to:

```
// imagine appCfg to be *fleet.AppConfig
var clonedAppCfg fleet.AppConfig
clonedAppCfg = *appCfg
return &clonedAppCfg
```

I tried `CopyWithOption(..., copier.Option{DeepCopy: true, IgnoreEmpty: true})` but it broke a bunch of tests. I'll take another look without the `IgnoreEmpty`, but I don't want to waste too much time debugging a third-party library.

Reply:
We were using an old-ish version of the `copier` package, updating it and using `CopyWithOptions(..., copier.Option{DeepCopy: true, IgnoreEmpty: true})` now works as expected, makes the code that much simpler.